### ISSUE TITLE
Issue in kernel what prevent using asan flags with gcc or clang

### ISSUE THREAD

**Issue by 4144** [coreos/bugs#2105]
# Issue Report #

## Bug ##

In kernel 4.11.12 was added some changes what prevent working [address sanitizer](https://github.com/google/sanitizers/wiki/AddressSanitizer) with gcc or clang.

for fix this issue already created patch here https://marc.info/?l=linux-kernel&m=150213695826001&w=2

Is possible to apply this patch to next coreos kernel update?

### Container Linux Version ###

Detected this issue in CI and cant say exact version where bug was introduced, but can show good and bad closest kernel versions:

kernel without issue:
```
Linux runner-4e4528ca-project-180398-concurrent-0 4.11.11-coreos #1 SMP Tue Jul 18 23:06:59 UTC 2017 x86_64 GNU/Linux
```
kernel with issue:
```
Linux runner-4e4528ca-project-180398-concurrent-0 4.11.12-coreos-r1 #1 SMP Fri Aug 11 21:51:18 UTC 2017 x86_64 GNU/Linux
```

### Environment ###

issue kernel related

### Expected Behavior ###

Address sanitizers should works in any kernel

### Other Information ###

Debian already patched own kernel with fix: https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=869090

More info about issue:

https://github.com/google/sanitizers/issues/837

https://bbs.archlinux.org/viewtopic.php?pid=1729746

https://gitlab.com/gitlab-com/support-forum/issues/2350

**Comment by bgilbert** [comment]
The problem was introduced by coreos/linux@9b1bbf6ea9b2b596ba271bec23b93c48181ad896.  The next stable release is due shortly and will have a 4.12 kernel, but that will not fix the problem.

**Comment by bgilbert** [comment]
The patch has been merged upstream as torvalds/linux@c715b72c1ba406f133217b509044c38d8e714a37, but is not yet in the stable queue.

**Comment by bgilbert** [comment]
Now in the 4.12 stable queue.

**Comment by bgilbert** [comment]
Now in 4.12.9.

**Comment by bgilbert** [comment]
Should be fixed in the next alpha release, 1520.0.0.  Backported to the beta and stable channels but not yet scheduled into a release on those channels.

**Comment by bgilbert** [comment]
Should be fixed in beta 1492.6.0 and stable 1465.7.0, due shortly.  Thanks for reporting.
